### ISSUE TITLE
Suggestions for Usability

### ISSUE THREAD

**Issue by solemcaelum** [cloudrex/discord-term#66]
Kind of defeats the purpose of having a Terminal Client, if you can't fully use said client in your terminal alone.

Keep in mind I'm on Windows 10, so I'm unable to click on channels. 

Some suggestions:
- Command/Hotkey to switch to most recent DM
- Some way to grab User IDs/see Friends list and IDs in client.
- Some way to grab Guild and Channel IDs in client. 
- Support Categories and Voice Channels (Command to see who's in voice)
- - Voice Support isn't necessary but would be neat if at all possible.

Feel free to let me know what doesn't work and what will work.

I'll also give the code a looksie later tonight and see about accomplishing some of the above and sending a PR.

**Comment by cloudrex** [comment]
I'm aware of the limitations present in Windows, so as you suggest a hotkey system and tools/commands to grab important info. such as friend lists, user IDs, etc. is certainly something this project is missing.

Currently the project is not being actively developed (due to other projects I'm working on) so I could really use some PRs (and possibly active contributors).

That being said, given the interest that this project has received I plan to eventually commit a large chunk of my time to advance the project in the areas that it lacks.

Something that I have planned is to divide the core functionality code into multiple "built-in" extensions and basically modularize any features, systems, etc. added afterwards, this way greatly simplifying the entry barrier for new contributors and making the project extension-based so developers/users can easily implement their own private functionality without having to be forced to contribute to the project themselves.

**Comment by solemcaelum** [comment]
I don't think it would be possible to implement voice, at least not without a helper application of some kind. 

Thank you for taking the time to look this all over! The reason I'm so nitpick-y and suggesting this much is because Discord-Term is a very useful tool for immersion when I'm developing or working on one of my many non-graphical machines. The ability to just stay logged in and up-to-date despite being on a terminal is something you'd think Discord would have capitalized on sooner. 

I gave the code a look, and it doesn't seem like it would be too hard to contribute. I'll try to keep things as organized and clean as possible when submitting my PRs :P

**Comment by cloudrex** [comment]
Glad to hear that! I've recently started updating the codebase on the `dev` branch which would eventually be merged onto `master`, so if you're planning on contributing make sure to checkout and explore the `dev` branch.
